{"schemaVersion":{"minor":3,"major":0,"patch":0},"metadata":{"role":"collectionGroup","roleHeading":"API Collection","modules":[{"name":"Amplify"}],"title":"AmplifyError Implementations"},"sections":[],"identifier":{"interfaceLanguage":"swift","url":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/PluginError\/AmplifyError-Implementations"},"hierarchy":{"paths":[["doc:\/\/amplify-swift.Amplify\/documentation\/Amplify","doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/PluginError"]]},"kind":"article","topicSections":[{"title":"Initializers","generated":true,"anchor":"Initializers","identifiers":["doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/PluginError\/init(errorDescription:recoverySuggestion:error:)"]},{"anchor":"Instance-Properties","title":"Instance Properties","identifiers":["doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/PluginError\/errorDescription","doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/PluginError\/isOperationCancelledError","doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/PluginError\/recoverySuggestion","doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/PluginError\/underlyingError"],"generated":true}],"variants":[{"traits":[{"interfaceLanguage":"swift"}],"paths":["\/documentation\/amplify\/pluginerror\/amplifyerror-implementations"]}],"references":{"doc://amplify-swift.Amplify/documentation/Amplify/PluginError/errorDescription":{"abstract":[{"type":"text","text":"A localized message describing what error occurred."}],"type":"topic","title":"errorDescription","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/PluginError\/errorDescription","url":"\/documentation\/amplify\/pluginerror\/errordescription","role":"symbol","fragments":[{"kind":"keyword","text":"var"},{"text":" ","kind":"text"},{"kind":"identifier","text":"errorDescription"},{"text":": ","kind":"text"},{"text":"ErrorDescription","kind":"typeIdentifier","preciseIdentifier":"s:7Amplify16ErrorDescriptiona"}],"kind":"symbol"},"doc://amplify-swift.Amplify/documentation/Amplify/PluginError/isOperationCancelledError":{"abstract":[],"role":"symbol","type":"topic","kind":"symbol","url":"\/documentation\/amplify\/pluginerror\/isoperationcancellederror","fragments":[{"text":"var","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"isOperationCancelledError"},{"text":": ","kind":"text"},{"text":"Bool","preciseIdentifier":"s:Sb","kind":"typeIdentifier"}],"title":"isOperationCancelledError","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/PluginError\/isOperationCancelledError"},"doc://amplify-swift.Amplify/documentation/Amplify/PluginError/init(errorDescription:recoverySuggestion:error:)":{"fragments":[{"kind":"identifier","text":"init"},{"text":"(","kind":"text"},{"text":"errorDescription","kind":"externalParam"},{"text":": ","kind":"text"},{"preciseIdentifier":"s:7Amplify16ErrorDescriptiona","text":"ErrorDescription","kind":"typeIdentifier"},{"kind":"text","text":", "},{"text":"recoverySuggestion","kind":"externalParam"},{"kind":"text","text":": "},{"preciseIdentifier":"s:7Amplify18RecoverySuggestiona","kind":"typeIdentifier","text":"RecoverySuggestion"},{"text":", ","kind":"text"},{"text":"error","kind":"externalParam"},{"text":": ","kind":"text"},{"kind":"typeIdentifier","text":"Error","preciseIdentifier":"s:s5ErrorP"},{"text":")","kind":"text"}],"title":"init(errorDescription:recoverySuggestion:error:)","kind":"symbol","type":"topic","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/PluginError\/init(errorDescription:recoverySuggestion:error:)","url":"\/documentation\/amplify\/pluginerror\/init(errordescription:recoverysuggestion:error:)","abstract":[{"type":"text","text":"AmplifyErrors must be able to be initialized from an underlying error. If an AmplifyError is created"},{"type":"text","text":" "},{"type":"text","text":"with this initializer, it must store the underlying error in the "},{"type":"codeVoice","code":"underlyingError"},{"type":"text","text":" property so it can be"},{"type":"text","text":" "},{"text":"inspected later.","type":"text"}],"role":"symbol"},"doc://amplify-swift.Amplify/documentation/Amplify":{"url":"\/documentation\/amplify","type":"topic","title":"Amplify","role":"collection","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify","kind":"symbol","abstract":[]},"doc://amplify-swift.Amplify/documentation/Amplify/PluginError/recoverySuggestion":{"role":"symbol","title":"recoverySuggestion","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/PluginError\/recoverySuggestion","kind":"symbol","abstract":[{"text":"A localized message describing how one might recover from the failure.","type":"text"}],"type":"topic","fragments":[{"text":"var","kind":"keyword"},{"kind":"text","text":" "},{"text":"recoverySuggestion","kind":"identifier"},{"kind":"text","text":": "},{"preciseIdentifier":"s:7Amplify18RecoverySuggestiona","text":"RecoverySuggestion","kind":"typeIdentifier"}],"url":"\/documentation\/amplify\/pluginerror\/recoverysuggestion"},"doc://amplify-swift.Amplify/documentation/Amplify/PluginError/underlyingError":{"abstract":[{"type":"text","text":"The underlying error that caused the error condition"}],"type":"topic","title":"underlyingError","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/PluginError\/underlyingError","url":"\/documentation\/amplify\/pluginerror\/underlyingerror","role":"symbol","fragments":[{"kind":"keyword","text":"var"},{"text":" ","kind":"text"},{"kind":"identifier","text":"underlyingError"},{"text":": ","kind":"text"},{"text":"Error","kind":"typeIdentifier","preciseIdentifier":"s:s5ErrorP"},{"kind":"text","text":"?"}],"kind":"symbol"},"doc://amplify-swift.Amplify/documentation/Amplify/PluginError":{"identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/PluginError","fragments":[{"text":"enum","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"PluginError"}],"title":"PluginError","type":"topic","url":"\/documentation\/amplify\/pluginerror","kind":"symbol","navigatorTitle":[{"text":"PluginError","kind":"identifier"}],"abstract":[{"text":"Errors associated with configuring and inspecting Amplify Plugins","type":"text"}],"role":"symbol"}}}